src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts

```diff
@@ -1002,6 +1002,21 @@ class ReaderToolsModel {
       return libsynphony.selectGPCWordsFromCache(desiredGPCs, knownGPCs, restrictToKnownGPCs, allowUpperCase, syllableLengths, selectedGroups, partsOfSpeech);
   }
 
+  static selectWordsFromAllowedLists(stageNumber: number): string[] {
+
+    var stages: ReaderStage[] = model.getSynphony().getStages(stageNumber);
+
+    var words: string[] = [];
+    for (var i=0; i < stages.length; i++) {
+      if (stages[i].allowedWords)
+        words = words.concat(stages[i].allowedWords);
+    }
+
+    words.sort();
+
+    return _.uniq(_.compact(words), true);
```

Comment:
Hmmm, that raises another point: have we ever discussed (mainly with @hatton) whether word comparison should be  case-sensitive? Some words (like names) are only valid if capitalized, many (in Roman) are OK capitalized at start of sentence, in IPA script case is fully meaningful...I would think in most scripts we'd want to use a case-insensitive comparison, and list all words in LC. The user will be annoyed if we highlight every sentence-initial word that doesn't have a capitalized form in the word list. This is not a spelling or grammar checker.